andela/rivendell-ah
# #159204692 Users should receive descriptive Signup/Registration validation errors

#### What does this PR do?
- This PR provides validation for input entered by users during signup. It makes use of validatorJs and some custom validation.
#### Description of Task to be completed?
- Install validatorjs dependencies
- Set valldation rules for each input entered by user
- Create custom/descriptive error messages for invalid inputs
- Create custom validation to check existing email and/or existing username entered by user  
#### How should this be manually tested?
- Run npm install
- Setup postgres (if not done already)
- Create a database on postgres
- Run sequelize db:migrate
- Use Postman to test the route http://localhost:EnterPortNumberhere/api/users. Ensure to enter values for username, email and password in the request body (Can enter invalid inputs to views descriptive error messages).
#### Any background context you want to provide?
- N/A
#### What are the relevant pivotal tracker stories?
#159204692

#### Screenshots (if appropriate)
<img width="1073" alt="screen shot 2018-07-30 at 6 31 34 pm" src="https://user-images.githubusercontent.com/26818808/43413118-0a921b16-9427-11e8-94db-cfc4a105c666.png">

#### Questions:
- N/A
